mmaction/datasets/pipelines/pose_loading.py

```diff
@@ -0,0 +1,666 @@
+import copy as cp
+import pickle
+
+import numpy as np
+from mmcv.fileio import FileClient
+from scipy.stats import mode
+
+from ..registry import PIPELINES
+from .augmentations import Flip
+
+
+@PIPELINES.register_module()
+class UniformSampleFrames:
+    """Uniformly sample frames from the video.
+
+    Required keys are "total_frames", "start_index" , added or modified keys
+    are "frame_inds", "clip_len", "frame_interval" and "num_clips".
+
+    Args:
+        clip_len (int): Frames of each sampled output clip.
+        num_clips (int): Number of clips to be sampled. Default: 1.
+        test_mode (bool): Store True when building test or validation dataset.
+            Default: False.
+        seed (int): The random seed used during test time.
+    """
+
+    def __init__(self, clip_len, num_clips=1, test_mode=False, seed=255):
+
+        self.clip_len = clip_len
+        self.num_clips = num_clips
+        self.test_mode = test_mode
+        self.seed = seed
+
+    def _get_train_clips(self, num_frames, clip_len):
+        """Uniformly sample indices for training clips.
+
+        Args:
+            num_frames (int): The number of frames.
+            clip_len (int): The length of the clip.
+        """
+
+        assert self.num_clips == 1
+        if num_frames < clip_len:
+            start = np.random.randint(0, num_frames)
+            inds = np.arange(start, start + clip_len)
+        elif clip_len <= num_frames < 2 * clip_len:
+            basic = np.arange(clip_len)
+            inds = np.random.choice(
+                clip_len + 1, num_frames - clip_len, replace=False)
+            offset = np.zeros(clip_len + 1, dtype=np.int64)
+            offset[inds] = 1
+            offset = np.cumsum(offset)
+            inds = basic + offset[:-1]
+        else:
+            bids = np.array(
+                [i * num_frames // clip_len for i in range(clip_len + 1)])
+            bsize = np.diff(bids)
+            bst = bids[:clip_len]
+            offset = np.random.randint(bsize)
+            inds = bst + offset
+        return inds
+
+    def _get_test_clips(self, num_frames, clip_len):
+        """Uniformly sample indices for testing clips.
+
+        Args:
+            num_frames (int): The number of frames.
+            clip_len (int): The length of the clip.
+        """
+
+        np.random.seed(self.seed)
```

Comment:
Why use seed here

Reply:
We want the testing to be deterministic. So we set a random seed here to make the behavior of random deterministic.

Reply:
The current way of setting seed will affect globally. Use np random generator to constraint the this seed local.

BTW there was deterministic flags in our testing script using mmcv, why doesn't that work?

This is going to resemble rl practices, where testing perf heavily relies on seeds :(

Reply:
`np.random.seed` seems to take effect globally. This may modify the random number seed set by mmcv at the beginning of the program